### DISCUSSION TITLE
Any breaking changes from 2.31 to 2.37?

### DISCUSSION THREAD

**Discussion by Fabhi** [prometheus/prometheus#12156] [Q&A] [answered]
We were planning to upgrade our Prometheus servers to 2.37, currently on 2.31. Were there any breaking changes introduced? I went through the Release Notes and they do not mention anything, yet it feels right to confirm with the community.

**Comment by roidelapluie** [comment, marked as answer]
While there haven't been any major breaking changes, there are several important updates that you should be aware of:

- In the 2.35.0 release, built with Go 1.18, TLS 1.0 and 1.1 are disabled by default on the client-side, and certificates signed with the SHA-1 hash function are rejected (excluding self-signed root certificates). You can override the TLS minimum version using the `min_version` parameter of `tls_config`.
- In the same release, promtool has a new flag `--lint` enabled by default for the commands check rules and check config. This results in a new exit code (3) for linter errors.
- In version 2.34.0, the Classic UI was removed, and tracing migrated from Jaeger to OpenTelemetry-based tracing.
- In version 2.32.0 the default max retry time for remote-write changed from 100ms to 5 seconds.

While these changes might not be breaking for most use cases, it's important to review them in the context of your specific setup.

**Comment by Fabhi** [comment]
Thank you for the detailed response!
